### PR TITLE
fix Issue 23055 - importC: using compound-literal array as pointer in CTFE gives 'dereference of invalid pointer'

### PR DESCRIPTION
**Pull request by WalterBright**
CTFE doesn't work with pointers in general, only in some special cases. This "fixes" the problem by generating an appropriate error message.

In the process of debugging this, I also extended the AST printer to handle ArrayLiteralExp better than generically.

### CODE REVIEW

**Comment by WalterBright** [comment]
Apparently I can't make it an error. Have to make it work!

**Comment by WalterBright** [comment]
People with dementia are far too easy to manipulate by an unscrupulous family member, friend, or associate.

**Comment by WalterBright** [comment]
This is ready to go.

**Review by RazvanN7 — APPROVED** [review]

File: compiler/src/dmd/dinterpret.d [hunk 1]
```diff
@@ -6177,6 +6184,20 @@ public:
             return;
         }
 
+        if (!lvalue && result.isArrayLiteralExp() &&
```
[the remark this answers is not in the document]
Comment (WalterBright) [reply compiler/src/dmd/dinterpret.d:6187]: the other use is in the debugging printf

**Pull request merged**